application/listeners/posthelpers.go

```diff
@@ -63,7 +63,73 @@ func sendNotificationRequestFromOpenToCommitted(template string, post m.Post) {
 	}
 }
 
-func sendNotificationRequestFromCommittedToAccepted(template string, post m.Post) {
+// Until we have status auditing history, we don't know who reverted the Post to `open` status.
+//  So, tell both the requester and provider about it.
+func sendNotificationRequestFromCommittedToOpen(template string, post m.Post, eData m.PostStatusEventData) {
+	postUsers := GetPostUsers(post)
+
+	oldProvider := m.User{}
+	if err := oldProvider.FindByID(eData.OldProviderID); err != nil {
+		domain.ErrLogger.Printf("error preparing '%s' notification for old provider id, %v ... %v",
+			template, eData.OldProviderID, err)
+	}
+
+	providerNickname := oldProvider.Nickname
+	providerEmail := oldProvider.Email
+
+	if providerNickname == "" {
+		providerNickname = "Unknown User"
+		providerEmail = "Missing Email"
+	}
+
+	// First notify requester
+	data := map[string]interface{}{
+		"uiURL":             domain.Env.UIURL,
+		"postURL":           domain.GetPostUIURL(post.Uuid.String()),
+		"postTitle":         post.Title,
+		"providerNickname":  providerNickname,
+		"providerEmail":     providerEmail,
+		"requesterNickname": postUsers.Requester.Nickname,
+		"requesterEmail":    postUsers.Requester.Email,
+	}
+
+	msg := notifications.Message{
+		Template: template,
+		Data:     data,
+		ToName:   postUsers.Requester.Nickname,
+		ToEmail:  postUsers.Requester.Email,
+	}
+	if err := notifications.Send(msg); err != nil {
+		domain.ErrLogger.Printf("error sending '%s' notification to old provider, %s", template, err)
+	}
+
+	// Now notify the old provider
+	if oldProvider.Nickname == "" {
+		return
+	}
+
+	data = map[string]interface{}{
+		"uiURL":             domain.Env.UIURL,
+		"postURL":           domain.GetPostUIURL(post.Uuid.String()),
+		"postTitle":         post.Title,
+		"providerNickname":  providerNickname,
+		"providerEmail":     providerEmail,
+		"requesterNickname": postUsers.Requester.Nickname,
+		"requesterEmail":    postUsers.Requester.Email,
+	}
+
+	msg = notifications.Message{
+		Template: template,
+		Data:     data,
+		ToName:   oldProvider.Nickname,
+		ToEmail:  oldProvider.Email,
+	}
```

Comment:
Does this all need to be repeated? Maybe just change the fields that need to be changed.

Reply:
Good catch